src/pages/views/components/reports/reportSummary/reportSummaryItems.test.tsx

```diff
@@ -1,30 +1,26 @@
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import React from 'react';
+import { FetchStatus } from 'store/common';
 import * as utils from 'utils/computedReport/getComputedReportItems';
 
 import { ReportSummaryItems, ReportSummaryItemsProps } from './reportSummaryItems';
 
 jest.spyOn(utils, 'getComputedReportItems');
 
 const props: ReportSummaryItemsProps = {
-  report: { data: [] },
-  idKey: 'date',
-  children: jest.fn(() => null),
-  t: jest.fn(v => `t(${v})`),
+  intl: {
+    formatMessage: jest.fn((m, v) => JSON.stringify(v)),
+  } as any,
+  children: jest.fn(() => 'children'),
+  status: FetchStatus.inProgress,
 };
 
-test('computes report items', () => {
-  shallow(<ReportSummaryItems {...props} />);
-  expect(utils.getComputedReportItems).toBeCalledWith({
-    report: props.report,
-    idKey: props.idKey,
-  });
-  expect(props.children).toBeCalledWith({ items: [] });
```

Comment:
mostly trying to test the result of the dom, but i think i'll be able to test this still

Reply:
actually, this was confusing to me as `ReportSummaryItemsProps` does not have `report` or `idKey`. I assumed it was outdated. I have a test checking the skeleton for in progress, this is a snap so it would test the full dom (no accessible/visible stuff for the user to see which is why i used a snap).. and then i tested that with status complete, we see the value that i created for the children function which wasn't there before.

Reply:
also, the fact that "children" shows up in the dom for the test below is proof that the updated mock children function is being called, and what we really want to test is that the dom gets updated with that for complete status.

Reply:
Not sure I fully agree. I don't know that the snapshot guarantees that any piece of code is called.

Reply:
testing-library focuses on testing what the user sees. any code that is being called that we want to test is being called for the purpose of manipulating the dom in some way, right?

if you look at this test in the main branch right now, you'll see errors on report and idKey as they are not properties on ReportSummaryItemsProps. The snap is showing the entire "skeleton" with no extra content, if something changes that test would fail. Then we are also testing that when the status completes, the "children" content gets added to the screen.

We can meet to discuss if you feel something is missing.

Reply:
I just don't see the snapshot as confirmation that a particular function (i.e., getComputedReportItems) was ever called, it's inferred. The test only fails if there are changes. It kinda feels like we're losing some unit tests.

Are you suggesting that toBeCalledWith cannot be called with react testing-library?

Reply:
I'm able to test it... I'm not sure why my IDE is complaining about report and idKey... maybe it is getting messed up by the way it is extending and then merging types, but I'm getting

Property 'report' does not exist on type 'ReportSummaryItemsProps'.  for report and idKey. Those props are in ComputedReportItemsParams which is extended by ReportSummaryItemsOwnProps which then gets merged into ReportSummaryItemsProps.

regardless of my IDE's complaints, it does seem to be working.. I'll add the test